catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -1323,6 +1350,24 @@ private CswRecordCollection queryProductById(String id) throws CswException {
         return cswRecordCollection;
     }
 
+    private String getRange(String rangeHeader) throws UnsupportedQueryException {
+        String response = null;
+
+        if (rangeHeader != null) {
+            if (rangeHeader.startsWith(BYTES_EQUAL)) {
+                String tempString = rangeHeader.substring(BYTES_EQUAL.length());
+                if (tempString.contains("-")) {
+                    response = tempString.substring(0, tempString.lastIndexOf("-"));
```

Comment:
So maybe change this up so instead of a `substring` you tokenize the input and take the first token as the `bytesToOffset` and then log the fact that an invalidly formatted range was passed in? Conversely, you could just throw an exception on a bad format.